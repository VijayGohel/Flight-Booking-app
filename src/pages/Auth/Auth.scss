@import '../../variables.scss';

.login-container {
  margin-top: 50px;
  text-align: center;
  background-color: $cardColor;
  padding: 20px;
  border-radius: 0.5rem;
}

.signup-form {
  .form-input {
    background-color: $inputColor;
    outline: none;
    border: none;
    border-radius: 0.5rem;
    padding: 0.8rem;
    margin: 10px 0;
    width: 100%;
    font-size: 14px;
  }

  .login-link {
    font-size: 14px;
    cursor: pointer;
  }

  .signup-btn {
    height: 2rem;
    width: 6rem;
    font-size: 14px;
  }

  .signin-google {
    width: 50%;
  }

  .pass-match {
    font-size: 14px;
    align-self: flex-end;
    color: red;
  }
}
