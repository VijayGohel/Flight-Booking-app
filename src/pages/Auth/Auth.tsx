import { useGoogleLogin } from '@react-oauth/google'
import { useState, useEffect } from 'react'
import './Auth.scss'
import axios from 'axios'
import { useDispatch } from 'react-redux'
import * as AuthAction from '../../actions/AuthAction'

const Auth = () => {
  const initialState = {
    firstName: '',
    lastName: '',
    email: '',
    password: '',
    confirmPass: '',
  }

  const [isSignup, setIsSignup] = useState(true)
  const [passMatched, setPassMatched] = useState(true)
  const [data, setData] = useState(initialState)
  const [user, setUser] = useState([] as any)
  const [profile, setProfile] = useState([])
  const dispatch = useDispatch()

  useEffect(() => {
    if (user) {
      axios
        .get(
          `https://www.googleapis.com/oauth2/v1/userinfo?access_token=${user.access_token}`,
          {
            headers: {
              Authorization: `Bearer ${user.access_token}`,
              Accept: 'application/json',
            },
          }
        )
        .then((res) => {
          setProfile(res.data)
        })
        .catch((err) => console.log(err))
    }
  }, [user])

  const handleChange = (e: any) => {
    setData({ ...data, [e.target.name]: e.target.value })
  }

  const handleSubmit = (e: any) => {
    e.preventDefault()

    if (isSignup) {
      if (data.password != data.confirmPass) 
        setPassMatched(false)
      else
        dispatch(AuthAction.singup(data) as any)
    } 
    else 
      dispatch(AuthAction.login(data) as any)
  }

  const resetForm = () => {
    setPassMatched(true)
    setData(initialState)
  }

  const login: any = useGoogleLogin({
    onSuccess: (codeResponse) => setUser(codeResponse),
    onError: (error) => console.log('Login Failed:', error),
  })

  return (
    <div className="container d-flex justify-content-center">
      <div className="card login-container">
        <h3 className="card-title">{isSignup ? 'Sign Up' : 'Login'}</h3>

        <form className="signup-form" onSubmit={handleSubmit}>
          {isSignup && (
            <div className="row">
              <div className="col-md-6 col-sm-12">
                <input
                  type="text"
                  placeholder="First Name"
                  value={data.firstName}
                  onChange={handleChange}
                  name="firstName"
                  id="first-name"
                  className="form-input"
                />
              </div>
              <div className="col-md-6 col-sm-12">
                <input
                  type="text"
                  placeholder="Last Name"
                  value={data.lastName}
                  onChange={handleChange}
                  name="lastName"
                  id="last-name"
                  className="form-input"
                />
              </div>
            </div>
          )}
          <div className="row">
            <div className="col-sm-12">
              <input
                type="text"
                placeholder="Email"
                value={data.email}
                onChange={handleChange}
                name="email"
                id="email"
                className="form-input"
              />
            </div>
          </div>

          <div className="row">
            <div className={`${isSignup ? 'col-md-6' : ''} col-sm-12`}>
              <input
                type="password"
                placeholder="Password"
                value={data.password}
                onChange={handleChange}
                name="password"
                id="pass"
                className="form-input"
              />
            </div>
            {isSignup && (
              <div className={`${isSignup ? 'col-md-6' : ''} col-sm-12`}>
                <input
                  type="password"
                  placeholder="Confirm Password"
                  value={data.confirmPass}
                  onChange={handleChange}
                  name="confirmPass"
                  id="confirm-pass"
                  className="form-input"
                />
              </div>
            )}
          </div>

          {!passMatched && (
            <span className="pass-match">*Password does not match!</span>
          )}

          <div
            className="login-link"
            onClick={() => {
              setIsSignup((prev) => !prev)
              resetForm()
            }}
          >
            <span>
              {isSignup
                ? 'Already have an account? Login!'
                : "Don't have an account? Signup!"}
            </span>
          </div>

          <div className="row justify-content-center mt-2">
            <button type="submit" className="btn signup-btn">
              {isSignup ? 'Sign Up' : 'Login'}
            </button>
          </div>
        </form>

        <hr />

        <div className="row justify-content-center">
          <a onClick={login} className="btn signin-google">
            <i className="fa-brands fa-google"></i>
            <div style={{ width: '10px' }}></div>
            <span>Login with Google</span>
          </a>
        </div>
      </div>
    </div>
  )
}

export default Auth
