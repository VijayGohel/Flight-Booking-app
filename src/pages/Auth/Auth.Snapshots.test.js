import renderer from 'react-test-renderer'
import Auth from './Auth'
import store from '../../store/ReduxStore'
import { Provider } from 'react-redux'
import { GoogleOAuthProvider } from '@react-oauth/google'

it('by default signup window should be displayed', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <GoogleOAuthProvider clientId={process.env.GOOGLEOAUTH_CLIENTID || ''}>
        <Auth />
      </GoogleOAuthProvider>
    </Provider>
  )
  expect(tree).toMatchSnapshot()
})
