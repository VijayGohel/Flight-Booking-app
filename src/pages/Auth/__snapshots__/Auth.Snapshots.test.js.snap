// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`by default signup window should be displayed 1`] = `
<div
  className="container d-flex justify-content-center"
>
  <div
    className="card login-container"
  >
    <h3
      className="card-title"
    >
      Sign Up
    </h3>
    <form
      className="signup-form"
      onSubmit={[Function]}
    >
      <div
        className="row"
      >
        <div
          className="col-md-6 col-sm-12"
        >
          <input
            className="form-input form-control"
            id="first-name"
            name="firstName"
            onChange={[Function]}
            placeholder="First Name"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="col-md-6 col-sm-12"
        >
          <input
            className="form-input"
            id="last-name"
            name="lastName"
            onChange={[Function]}
            placeholder="Last Name"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-12"
        >
          <input
            className="form-input"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-6 col-sm-12"
        >
          <input
            className="form-input"
            id="pass"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="col-md-6 col-sm-12"
        >
          <input
            className="form-input"
            id="confirm-pass"
            name="confirmPass"
            onChange={[Function]}
            placeholder="Confirm Password"
            required={true}
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        className="login-link"
        onClick={[Function]}
      >
        <span>
          Already have an account? Login!
        </span>
      </div>
      <div
        className="row justify-content-center mt-2"
      >
        <button
          className="btn signup-btn"
          type="submit"
        >
          Sign Up
        </button>
      </div>
    </form>
    <hr />
    <div
      className="row justify-content-center"
    >
      <button
        className="btn signin-google"
        disabled={false}
        onClick={[Function]}
      >
        <i
          className="fa-brands fa-google"
        />
        <div
          style={
            {
              "width": "10px",
            }
          }
        />
        <span>
          Login with Google
        </span>
      </button>
    </div>
  </div>
</div>
`;
